pkg/dns/dns.go

```diff
@@ -420,8 +420,20 @@ func (kd *KubeDNS) newHeadlessService(service *kapi.Service) error {
 func (kd *KubeDNS) Records(name string, exact bool) (retval []skymsg.Service, err error) {
 	glog.Infof("Received DNS Request:%s, exact:%v", name, exact)
 	trimmed := strings.TrimRight(name, ".")
-	segments := strings.Split(trimmed, ".")
-	path := reverseArray(segments)
+	path := strings.Split(trimmed, ".")
+	isFederationQuery := false
+	federationPath := []string{}
+	if !exact && kd.isFederationQuery(path) {
+		glog.Infof("Received federation query. Going to try to find local service first")
+		// Try quering the non-federation (local) service first.
+		// Will try the federation one later, if this fails.
+		isFederationQuery = true
+		federationPath = append(federationPath, path...)
+		// To try local service, remove federation name from path.
+		// Federation name is 3rd in the path (after service name and namespace).
+		path = append(path[:2], path[3:]...)
```

Comment:
It works when there is no endpoint for the local service (I have a test for it).
But I just tried deleting endpoints from an existing service and it still returned me the endpoint (which I deleted). Am debugging that.